src/client/camera.cpp

```diff
@@ -533,7 +533,20 @@ void Camera::update(LocalPlayer* player, f32 frametime, f32 busytime, f32 tool_r
 	m_wieldnode->setPosition(wield_position);
 	m_wieldnode->setRotation(wield_rotation);
 
-	m_wieldnode->setColor(player->light_color);
+	ItemStack wielditemstack;
+	ItemDefinition wielditemdef;
+	IItemDefManager *idef = m_client->getItemDefManager();
+
+	wielditemstack = player->getWieldedItem(&wielditemstack, NULL);
+	wielditemdef = wielditemstack.getDefinition(idef);
+
+	u8 light = decode_light(wielditemdef.light_source); // Not sure if I need to check wether light_source is in the def
+	video::SColor wieldlight(255,light,light,light);
+
+	if (wielditemdef.light_source > 0 && wieldlight.getLuminance() > player->light_color.getLuminance())
+		m_wieldnode->setColor(wieldlight);
+	else
+		m_wieldnode->setColor(player->light_color);
```

Comment:
```cpp
    if (wielddef.light_source > 0 && light > player->light_color.getLuminance())
        m_wieldnode->setColor(video::SColor(255, light, light, light));
    else
        m_wieldnode->setColor(player->light_color);
```

Reply:
Thanks, I think I had some missing code that lead me to believe that I couldn't compare `light` with the value returned by `getLuminance()`

Reply:
I'll let you close this if it's resolved